src/actions/rsaa/headers.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash';
+
+/**
+ * Check if headers object supports entities() method.
+ * @param {Headers} headers
+ */
+function isHeadersEntriesSupported(headers) {
+  return (_.isFunction(headers.entries));
+}
+
+/**
+ * Return existing or create new Headers object instance
+ * @param {Headers|Object} headers
+ */
+function getHeadersInstance(headers) {
+  if (headers instanceof Headers) {
+    return headers;
+  }
+
+  return new Headers({ ...headers });
+}
+
+/**
+ * Extract Request/Response headers into a plain object
+ * @param {Object} response
+ */
+export function extractHeaders(resource) {
+  if (!_.has(resource, 'headers')) {
+    return {};
+  }
+
+  const headersInstance = getHeadersInstance(resource.headers);
+  const headers = {};
+  let currentHeader = null;
+
+  if (isHeadersEntriesSupported(headersInstance)) {
+    const iterator = headersInstance.entries();
+    for (currentHeader = iterator.next(); !currentHeader.done; currentHeader = iterator.next()) {
```

Comment:
for .. of, as previous comment suggested. Please check does it trasnpile into universal supported code.
https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Cstage-0&targets=&browsers=&builtIns=false&debug=false&code_lz=GYewTgFAhgBCwwEYEoYG8BQMYGMQDsBnEAGwFMA6EkAcwgHIomp7kBuDAXyA